enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,501 @@
+    - Configure Egress rules in the Security Group to allow traffic to the backend pods on the targetPort specified in the Service's `spec.ports` and the health check port. Initially, this should be restricted to the cluster's VPC CIDR or the specific CIDRs of the worker nodes.
+    - When creating the NLB using the AWS ELBv2 API, the CCM will include the ID of the newly created Security Group in the `SecurityGroups` parameter of the `CreateLoadBalancerInput.`
+  - When the Service annotation is added after the Service is created, the CCM will (?)
+  - When the Service is deleted, the CCM will also delete the associated Security Group, ensuring proper cleanup.
+
+#### OpenShift Managed (TBD)
+
+- ROSA Classic:
+    - TBD: Ensure the `install-config.yaml` option correctly writes the CIO manifests enabling NLB with managed Security Groups. This might involve changes in the Hive component responsible for cluster provisioning.
+
+- ROSA HCP:
+    - TBD: Investigate how Hypershift sets the Service Annotation of Security Group management when launching CIO. The `install-config.yaml` option should influence the Hosted Control Plane's CIO configuration. Need to validate if ROSA HCP follows the pattern of Hypershift or require additional configuration.
+
+### API Extensions
+
+> WIP/TBReviewed
+
+#### AWS Cloud Controller Manager (CCM)
+
+- Introduction of a new annotation constant: `ServiceAnnotationLoadBalancerManagedSecurityGroup = "service.beta.kubernetes.io/aws-load-balancer-managed-security-group"` in `pkg/providers/v1/aws.go`.
+
+```go
+// ServiceAnnotationLoadBalancerManagedSecurityGroup is the annotation used
+// on the service to specify the instruct CCM to manage the security group when creating a Network Load Balancer. When enabled,
+// the CCM creates the security group and it's rules. This option can not be used with annotations
+// "service.beta.kubernetes.io/aws-load-balancer-security-groups" and "service.beta.kubernetes.io/aws-load-balancer-extra-security-groups".
+const ServiceAnnotationLoadBalancerManagedSecurityGroup = "service.beta.kubernetes.io/aws-load-balancer-managed-security-group"
+```
+
+- Logic in the service controller within the CCM (`pkg/providers/v1/aws.go` and `pkg/providers/v1/aws_loadbalancer.go` ) to recognize and handle the new annotation when the service type is `NLB` (`ServiceAnnotationLoadBalancerType = "service.beta.kubernetes.io/aws-load-balancer-type"` ).
+- Functionality within the CCM to create and manage the lifecycle of AWS Security Groups for NLBs, including creating ingress and egress rules based on the service specification. This would likely involve using the AWS SDK for Go to interact with the EC2 API for creating and managing security groups.
+
+#### Cluster Ingress Operator (CIO)
+
+- FeatureGate TechPreview (TODO describe the name?)
+- Introduce a new boolean field `managedSecurityGroup` in the `NetworkLoadBalancer` provider parameters within the IngressController API (`operator.openshift.io/v1`).
```

Comment:
We don't tend to use boolean fields, I suspect you want an API that looks more like
```
parameters:
  securityGroups:
    type: None | Managed | Unmanaged
    unmanaged:
      securityGroupIDs: []
```
Which would allow someone to specify they want no security group (not sure if we actually want this long term?) or a managed security group, or to attach specific security groups

Reply:
Closing this thread as we are not planning to change CIO API